### ISSUE TITLE
ToImage would raise panic: runtime error: index out of range.

### ISSUE THREAD

**Issue by lbeeon** [lazywei/go-opencv#99]
code blow:
```
	filename := "./test.jpeg"
	img := opencv.LoadImage(filename, opencv.CV_LOAD_IMAGE_UNCHANGED)
	if img == nil {
		fmt.Printf("LoadImage fail: %s\n", filename)
		return
	}
	srcImage := img.ToImage()
	cvImage := opencv.FromImage(srcImage)
	srcImage = cvImage.ToImage()
	imaging.Save(srcImage, "cvout.jpeg")
	defer img.Release()

```

Any idea?

**Comment by lbeeon** [comment]
```
/* ToImage converts a opencv.IplImage to an go image.Image */
func (img *IplImage) ToImage() image.Image {
	var height, width, channels, step int = img.Height(), img.Width(), img.Channels(), img.WidthStep()
	out := image.NewNRGBA(image.Rect(0, 0, width, height))
	if img.Depth() != IPL_DEPTH_8U {
		return nil // TODO return error
	}
	// Turn opencv.Iplimage.imageData(*char) to slice
	var limg *C.char = img.imageData
	var limg_ptr unsafe.Pointer = unsafe.Pointer(limg)
	var data []C.char = (*[1 << 30]C.char)(limg_ptr)[:height*step : height*step]

	c := color.NRGBA{R: uint8(0), G: uint8(0), B: uint8(0), A: uint8(255)}
	// Iteratively assign imageData's color to Go's image
	for y := 0; y < height; y++ {

>  step should increase 4?
>  otherwise the channel a would be assign to next pixel channel b

		for x := 0; x < step; x = x + 3 {
			c.B = uint8(data[y*step+x])
			c.G = uint8(data[y*step+x+1])
			c.R = uint8(data[y*step+x+2])
			if channels == 4 {
				c.A = uint8(data[y*step+x+3])
			}
			out.SetNRGBA(int(x/3), y, c)
		}
	}

	return out
}

```

**Comment by zacharynevin** [comment]
I made a PR for this

**Comment by zacharynevin** [comment]
https://github.com/lazywei/go-opencv/pull/108

**Comment by zacharynevin** [comment]
#108

**Comment by lazywei** [comment]
Closed in #108
